FreeIT/src/lesson_10/Application.java

```diff
@@ -0,0 +1,237 @@
+
+public class Application {
+
+	private Library libr;
+
+	public Application() {
+		super();
+		ArrayList<Book> book = new ArrayList<>();
+		Book book1 = new Book(123, "Atlant", "Roman");
+		Book book2 = new Book(345, "Puaro", "Detective");
+		Book book3 = new Book(567, "NLO", "Fantasy");
+		book.add(book1);
+		book.add(book2);
+		book.add(book3);
+		Library lib = new Library(book);
+		libr = lib;
+	}
+
+	public void start() {
+		menu();
+	}
+
+	private void menu() {
+		System.out.println(
+				"Выберите действие (введите число, соответствующее действию): \n1. Вывод всех книг.\n2. Добавление книги.\n3. Удаление книги.\n4. Редактирование книги.\n5. Выход.");
+		Scanner sc = new Scanner(System.in);
+		int choice = 0;
+		try {
+			choice = sc.nextInt();
+		} catch (InputMismatchException e) {
+			System.out.println("Вы должны ввести число! Попробуйте еще раз!");
+			menu();
+			return;
+		}
+		switch (choice) {
+		case 1:
+			printAllBooks();
+			menu();
+			break;
+		case 2:
+			addingBook();
+			menu();
+			break;
+		case 3:
+			deletingBook();
+			menu();
+			break;
+		case 4:
+			correctingBook();
+			menu();
+			break;
+		case 5:
+			System.out.println("До свиданья!");
+			break;
+		default:
+			System.out.println("Число введено неверно. Попробуйте еще раз!");
+			menu();
+		}
+
+	}
+
+	private void printAllBooks() {
+		if (libr.getBooks() == null) {
+			System.out.println("В библиотеке нет книг. Нечего выводить!");
+		} else {
+			System.out.println(
+					"Выберите желаемую сортировку при выводе (введите число, соответствующее желаемой сортировке): \n1. По алфавиту по названию книги.\n2. По добавлению(сначала новые, потом более старые).\n3. По добавлению(сначала старые, потом новые).");
+			Scanner sc = new Scanner(System.in);
+			int choice = 0;
+			try {
+				choice = sc.nextInt();
+			} catch (InputMismatchException e) {
+				System.out.println("Вы должны ввести число! Попробуйте еще раз!");
+				printAllBooks();
+				return;
+			}
+			switch (choice) {
+			case 1:
+				sort1();
+				System.out.println("Вывод книг прошел успешно!");
+				menu();
+				break;
+			case 2:
+				sort2();
+				System.out.println("Вывод книг прошел успешно!");
+				menu();
+				break;
+			case 3:
+				sort3();
+				System.out.println("Вывод книг прошел успешно!");
+				menu();
+				break;
+			default:
+				System.out.println("Число введено неверно. Попробуйте еще раз!");
+				printAllBooks();
+			}
+		}
+
+	}
+
+	private void sort1() {
+		List<Book> a = libr.getAll();
+		a.sort(new Comparator() {
+
+			@Override
+			public int compare(Object o1, Object o2) {
+				Book b1 = (Book) o1;
+				Book b2 = (Book) o2;
+				String s1 = b1.getTitle();
+				return s1.compareTo(b2.getTitle());
+			}
+		});
+		System.out.println(a);
+	}
+
+	private void sort2() {
+		System.out.println(libr);
+	}
+
+	private void sort3() {
+		ListIterator<Book> iter1 = libr.getBooks().listIterator();
+		ArrayList<Book> b1 = new ArrayList<>();
+		while (iter1.hasNext()) {
```

Comment:
you don't need go forward and then backward just to print list. If you need just print it - go from back to start without creating additional difficulties

Reply:
How to assign the iterator of ArrayList to the last element? Should I change it to LinkedList?